README.md

```diff
@@ -12,61 +12,80 @@ letting user choose from a wide selection of predefined Dockerfile snippets
 NPM installs).
 
-### Build
+## Build
 Make sure you setup Go acording to: https://golang.org/doc/install#install
 ```
 go get -v
 go build
 ```
 
-### Usage
+## Usage
 
+Basicaly, pazuzu cli tool has 3 steps:
+- **search** - search for available features
+- **compose** - compose **Pazuzufile** with desired features
+- **build** - create Dockerfile out of Pazuzufile
+
+## Search features
+List step is used to check what features are actually available in configured snippet provider
+```
+$ pazuzu search [regexp]
 ```
-pazuzu -h
-NAME:
-   pazuzu - Build Docker features from pazuzu-registry
-
-USAGE:
-   pazuzu [global options] command [command options] [arguments...]
-
-VERSION:
-   0.1
-
-COMMANDS:
-     build   build docker image
-     verify  verify docker image against
-     search  search for features in registry
-     list    list all features in registry
-
-GLOBAL OPTIONS:
-   --docker-endpoint value, -e value     Set the docker endpoint (default: "unix:///var/run/docker.sock")
-   --registry value, -r value            Set the registry URL (default: "http://localhost:8080/api") [$PAZUZU_REGISTRY]
-   --tokeninfo-endpoint value, -t value  Sets the OAuth2 token info URL (default: "https://token.auth.zalando.com/access_token")
-   --user value, -u value                Sets the OAuth2 user name
-   --help, -h                            show help
-   --version, -v                         print the version
+Mask is a valid regexp, i.e.:
 ```
+$ pazuzu search "node-v4.6*|java8|mvn"
+```
+Mask `node-v4.6*|java8|mvn` will use following features:
+- latest `node` version that match regexp **node-v4.6***
+- latest `java8` release available in feature snippets provider(default java8 )
 
+## Compose features
+Compose step actually creates features file out of specified 
+```
+$ pazuzu compose <space-separated-feature-names> 
 ```
-pazuzu build -h
+### Available options
+- **--with-base NAME** - specify base image
+- **-o/--out FILENAME** - specify name of the output features file. Default is Pazuzu file
+### Examples:
+`$ pazuzu compose node java8` - generate Pazuzufile from the list of features
 
-NAME:
-   pazuzu build - build docker image
+`$ pazuzu compose node java8 --out node-und-java8.yml` - Save everything to **node-und-java8.yml** file
 
-USAGE:
-   pazuzu build [command options] [arguments...]
+## Build Dockerfile
+This step aims to actually create **Dockerfile** out of the snippets configured for features.
+
+`$ pazuzu build . `  - Builds a **Dockerfile** from **Pazuzufile** located in the current directory.
+
+` $ pazuzu build <node-with-babel.yml>` - Builds a **Dockerfile** from specified feature file.
+
+**NOTE:** build command gives a sample command of how to run `docker
+build` 
+
+## Run docker build
+Execute `docker build` command to actually create image
+
+---
+## Installation and Configuration
+All set configuration will be stored ` ~/.pazuzu_conf`
+
+-  Setup snippets provider:
+
+    ```
```

Comment:
I need to check with @stoewer what parameters he actually needs for the git library to connect. Then it will be added to README